### DISCUSSION TITLE
dark mode is missing in transmission-gtk 4.0.1

### DISCUSSION THREAD

**Discussion by BobRock55** [transmission/transmission#5140] [Q&A]
### What is the issue?

Updated to ver 4.0.1 (e1c6e1be43) and now the dark mode is gone making this app to be the only one in white and hurting my eyes. Can it be set manually or wait for the fix?
Thanks

### Which application of Transmission?

None

### Which version of Transmission?

4.0.1 (e1c6e1be43)

**Comment by ckerr** [comment]
>> Which application of Transmission?

> None.

Web UI? Qt? Mac? GTK? What desktop are you using?

I don't understand what answers you want if you won't fill out the issue template.

**Comment by BobRock55** [comment]
Sorry, first time posting... Desktop Linux (Pop!_os ver. 22.04)

**Comment by GaryElshaw** [comment]
Were you meaning the web client (if so, which browser?), the GTK or Qt client?

**Comment by BobRock55** [comment]
No browser, stand alone app available at Linux shop repository  (guess native Linux interface).

**Comment by nevack** [comment]
> No browser, stand alone app available at Linux shop repository (guess native Linux interface).

Please, provide screenshot of about window.

**Comment by BobRock55** [comment]
Hope this helps.
![pic](https://user-images.githubusercontent.com/86332222/222876987-14eba165-c6e1-44e1-af9c-3b6d558bf5df.png)

**Comment by bbhtt** [comment]
What theme are you using? And is this Flatpak or from your OS repository? If it is Flatpak (it is built against GTK4) but the flatpaked version of PopOS GTK theme doesn't support GTK4 applications see https://github.com/pop-os/gtk-theme/issues/553. If it is some other theme, make sure it supports GTK4 and has a dark style.

You can 'force' dark theme on it by using the environment variable `GTK_THEME=Adwaita:dark` or by changing the desktop theme to that in GNOME Tweaks.

**Comment by BobRock55** [comment]
Thank you for the detailed answer. I had dark mode until update this afternoon. 
Changed to Adwaita:dark and is working fine (was Pop:dark before)
![pic](https://user-images.githubusercontent.com/86332222/222877880-6a533c08-62fd-4936-94e5-7378381e93bc.png)

**Comment by ckerr** [comment]
I'm going to convert this to a discussion since it's clearly a real issue users should be able to search for and find discussion on, but it doesn't seem to be a bug in Transmission's code.

**Comment by bojan-stojanovic** [comment]
I think the problem is that Transmission 4 doesn't inherit global dark theme. If I set environment variable as mentioned before, it is true that Transmission will use dark theme, but it will break look of other gnome apps.

OS: Arch linux - Gnome 43
App: transmission-gtk (from repositories)

![image](https://user-images.githubusercontent.com/32634808/222960224-e81d433e-1f33-4eb8-8fcb-99089fc94315.png)

**Reply by bbhtt** [reply]
Yes it doesn't use libadwaita but uses GTK4.

> Transmission will use dark theme, but it will break look of other gnome apps.

Set it in the launcher or desktop entry.

**Comment by rail01** [comment]
I use [adw-gtk3](https://github.com/lassekongo83/adw-gtk3) on GNOME 43.3 and Arch Linux using `transmission-gtk` 4.0.1-2 from the repo and get the same problem. Very annoying.
Opening the app with `GTK_THEME=Adwaita:dark` makes it look terrible and inconsistent, which is exactly the issue I aimed as fixing with making GTK3 apps look like Libadwaita.
![Screenshot from 2023-03-07 20-53-40](https://user-images.githubusercontent.com/29921838/223537770-a5b17917-a8c3-400f-aae8-9753660a7223.png)

**Reply by bbhtt** [reply]
That theme is for GTK3 applications only, since Transmission is GTK4 it doesn't get applied to it. Follow https://github.com/lassekongo83/adw-gtk3/blob/main/gtk4.md for a workaround.

It looks mostly ok with a few inconsistencies:
 
![image](https://user-images.githubusercontent.com/62639087/223597589-e47b19a9-0799-41d2-a846-3ec9a87d3672.png)

**Comment by rail01** [comment]
@bbhtt huh, I assumed the inconsistencies come from Transmission still using GTK3. It being GTK4 but not Libadwaita is frankly even worse :/

**Comment by vitaly-zdanevich** [comment]
No way it can be fixed in Transmission?

**Comment by ExecutorElassus** [comment]
How would you set the theme in a non-GNOME WM? I'm using fvwm. Setting GTK_THEME in my .bashrc doesn't seem to work for transmission.

**Reply by mikedld** [reply]
I'm using [xsettingsd](https://codeberg.org/derat/xsettingsd) which works well for both GTK 3 and 4.

**Reply by o-alquimista** [reply]
It would need to be set and exported in your login shell (`.bash_profile`), then log out.

```sh
export GTK_THEME="Adwaita:dark"
```

**But don't do that**, because it breaks the appearance of libadwaita apps.

# Solution: override .desktop file
Instead, override the Transmission `.desktop` file, effectively restricting the effects of `GTK_THEME` to Transmission alone.

```sh
# Make a copy of the .desktop file for your user
cp /usr/share/applications/transmission-gtk.desktop ~/.local/share/applications/
```

```diff
-Exec=transmission-gtk %U
+Exec=env GTK_THEME=Adwaita:dark transmission-gtk %U
```

(do the same for the other Exec lines in the file; i.e. other "actions")

# Alternative solution: _gtk-application-prefer-dark-theme_
Follow [Derecho's solution](https://github.com/transmission/transmission/discussions/5140#discussioncomment-6305849).

> It suffices (for me) to simply create a `~/.config/gtk-4.0/settings.ini`.

```ini
[Settings]
gtk-application-prefer-dark-theme=true
```

Restart Transmission and it should be dark.

This solution does not appear to break libadwaita apps, but beware it will apply to any application using GTK4 (instead of libadwaita) like Transmission is.

___

I'm also using _xsettingsd_.

**Comment by kanehekili** [comment]
Having a similar problem with TransmissionQt on XFCE. The theme defines a GTK2 theme. Using Qt5-styleplugins. Environment set to 

- QT_QPA_PLATFORMTHEME=gtk2
- QT_STYLE_OVERRIDE=gtk2

If TransmissionQT looses focus (GTK3 backdrop behaviour), the combo boxes look like this:
![image](https://github.com/transmission/transmission/assets/4390452/59714284-dc2d-4e61-b335-0dd498e12b5a)

The bars turned from blue to green (which I don't mind -probably correct) but the combo boxes are broken. Other QT Apps on this system display them right.

System is a Manjaro 23.0.0 UltimaThule, Kernel is  x86_64 Linux 6.1.31-2-MANJARO. 

Note that the combo boxes are displayed correctly, if the app has the focus. It is only if they **don't ** have it.  Checked with my theme (https://www.gnome-look.org/p/1424967/) as well as Adwaita-dark

**Comment by Derecho** [comment]
I've noticed the same issue of Transmission GTK suddenly not setting dark mode anymore after having had it in a previous version. As I don't run gnome nor do I have a replacement for handling xsettings, I was using `~/.config/gtk-3.0/settings.ini` to set `gtk-application-prefer-dark-theme = true`.

From this topic I gather that Transmission is now built against gtk 4.0 rather than 3.0, so this file is no longer read and applied. The setting however is still valid in gtk 4.0: https://docs.gtk.org/gtk4/property.Settings.gtk-application-prefer-dark-theme.html

It suffices (for me) to simply create a `~/.config/gtk-4.0/settings.ini` and fill it with the very same contents that I used for gtk3. (I created the dir and copied the file). Restarting Transmission made it show up dark again, as I was previously used to, resolving this issue.

**Reply by o-alquimista** [reply]
It works for me too. Did you have any problems with other apps using this setting?

**Reply by dreua** [reply]
Applied that workaround on Fedora 42 today, the main Gnome setting used to work I think, but a few weeks ago some apps including Transmission switched back to light theme. Can't figure out why.

**Comment by ExecutorElassus** [comment]
Derecho, can you post the content of your settings.ini file? I copied mine over from the gtk-3.0 directory, but don't see any change in transmission's theme.

**Reply by Derecho** [reply]
It's no more than this really:
```
[Settings]
gtk-application-prefer-dark-theme = true
```

**Comment by kanehekili** [comment]
Well, it's dark now, but searchbar looks like this:
![image](https://github.com/transmission/transmission/assets/4390452/f1396fbc-b2e6-4a06-bf21-ecb1458e5b21)

**Comment by mathiascode** [comment]
This could be resolved by implementing the Freedesktop color-scheme preference. Example implementations:
https://github.com/nicotine-plus/nicotine-plus/blob/24aa29a7aff16cfc1afd346de49accd500eb3dc9/pynicotine/gtkgui/widgets/theme.py#L47-L94
https://gitlab.gnome.org/GNOME/evolution/-/blob/master/src/e-util/e-color-scheme-watcher.c

Alternatively, you can use libadwaita, which supports this out of the box. It's probably not the direction Transmission wants to go in, but you could make libadwaita optional and only enable it when using GNOME.

**Comment by kanehekili** [comment]
Update - without remembering to have changed anything, Transmission-qt displays correctly with qt5ct set to qtk2 and/or setting QT_QPA_PLATFORMTHEME=gtk2. Issue should be closed.

**Comment by AuditeMarlow** [comment]
Is this feature available for the Windows builds as well?

**Reply by kanehekili** [reply]
Nah, my XFCE desktop used GTK3/4 by default, your Windoze needs to import the QT6 libraries. Completely different setup. So there is no "translation from GTK to QT theme" in your environment.

**Reply by AuditeMarlow** [reply]
I more or less meant to ask if dark mode is on Windows builds at all?

**Reply by mikedld** [reply]
This was discussed a number of times in issues, most recently in #6469.

**Comment by Together34** [comment]
Setting "Legacy applications" to "Adwaita-Dark" in Gnome Tweaks has solved the issue in an acceptable manner for me, without causing any surprises so far in other apps. 

- Ubtuntu 23.10
- GNOME Shell 45.2

Not sure what other info might be useful.

![Screenshot from 2024-03-11 21-51-01a](https://github.com/transmission/transmission/assets/58236114/6218dbfa-3055-4c2e-9b7a-79c301dc863c)


![Screenshot from 2024-03-11 22-03-28](https://github.com/transmission/transmission/assets/58236114/2624b32b-b349-4878-bbc8-0e71d11dbe8a)

**Reply by GaryElshaw** [reply]
Please mask or delete any images containing copyrighted material.

**Reply by Together34** [reply]
Aye sir!

I hope this is acceptable, will fix in completely "safe way", if still poses risks/problems.

**Comment by haevalencia** [comment]
Hello, I recently installed fedora workstation 40 vanilla and transmission is the only app I use that does not change to dark mode automatically. From the things posted in this discussion, I used the gnome-tweak option of using adwaita-dark in legacy applications, but that breaks other apps by not changing the theme automatically. Could you describe the method to use libadwaita?
As a user I wouldn't expect much attention to something that only "apparently" is a gnome thing to work by default, but an integration option in the app's preferences would be appreciated to be able to handle it easily. Greetings.

**Reply by mattismyname** [reply]
Agreed, it does seem very strange that Transmission is the only common app I use that doesn't just use dark mode out of the box on Fedora.

**Comment by kanehekili** [comment]
I'd like to chime in. I am using transmission-qt in a GTK environment (XFCE) but do not have any problems with a dark theme. 
I am using my [own dark theme](https://www.pling.com/p/1424967/) which includes a gtk2 css file. In addition with qt6-gtk2 (on Arch) and `QT_QPA_PLATFORMTHEME=gtk2` in /etc/environment the qt Version should pick up the theme. On the page mentioned above  I elaborated how to get that theme running for QT applications - it should work for all those themes that have the gtk2 part delivered.
